File: app/src/main/java/cat/xlagunas/andrtc/di/components/ConferenceComponent.java
package cat.xlagunas.andrtc.di.components;

import cat.xlagunas.andrtc.di.ConferenceScope;
import cat.xlagunas.andrtc.di.modules.ConferenceModule;
import cat.xlagunas.andrtc.view.activity.ConferenceActivity;
import dagger.Subcomponent;

/**
 * Created by xlagunas on 4/8/16.
 */
@ConferenceScope
@Subcomponent(modules = {ConferenceModule.class})
public interface ConferenceComponent {
    void inject(ConferenceActivity activity);
}

File: conference/src/main/java/cat/xlagunas/conference/di/ConferenceComponent.kt
package cat.xlagunas.conference.di

import cat.xlagunas.conference.ui.ConferenceActivity
import cat.xlagunas.viv.dagger.ApplicationComponent
import dagger.BindsInstance
import dagger.Component
import cat.xlagunas.core.di.Feature

@Component(modules = [ConferenceModule::class], dependencies = [ApplicationComponent::class])
@Feature
interface ConferenceComponent {
    @Component.Builder
    interface Builder {
        fun parent(component: ApplicationComponent): Builder
        @BindsInstance
        fun roomId(roomId: String): Builder

        fun build(): ConferenceComponent

        @BindsInstance
        fun activity(activity: ConferenceActivity): Builder
    }

    fun inject(activity: ConferenceActivity)
}